docs/version_control.md

```diff
@@ -24,23 +29,60 @@ Nearly all operations that are performed by Git are in you local computing envir
 | git push                      | upload staged commit to repo                                       |
 | git pull                      | get remote repo commits and download \(try and resolve conflicts\) |
 | git clone &lt;url&gt;         | download entire repository                                         |
+| git checkout &lt;branch&gt;   | checkout and create the branch you want to use                                | 
+  
+  
+```bash
+# clone project 
+git clone https://github.com/brown-ccv/honeycomb.git
+# create branch with your feature
+git checkout -b feature_name
+# check the status of your repositoey
+git status
+# commit file contents to the local repository
+git commit -am "My feature is ready"
+# specific message
+# push file contents to the remote (i.e. cloud) repository
+git push origin feature_name
+```
+### Alternative options 
+  
+Instead of using commands in the terminal, you can also download [GitHub desktop](https://desktop.github.com/). It is very intuitive to use. 
+  
+#### Clone repository 
+
+![](assets/git_clone.png)
+  
+#### Select a branch or create a new branch
+
+![](assets/git_branch.png)
+  
+#### Commit changes and push
+
+![](assets/git_commit.png)
+
+## Create a Pull Request 
+
+Pull requests are useful before you merge your branch with the main branch. You can request a review from your colleagues and check for any conflicts with the main branch. After you pushed all the changes to your branch, you can go to the original GitHub repository and click on the pull request.  
+
+![](assets/pull_request.png)
+
+![](assets/pull_request_info.png)
 
 ## Best Practices
 
-### Workflow
+ ### Git Workflow
+  
+ We recommend using a simple flow based on following rules:
 
-We recommend a simple flow based on following rules:
+- Use topic/feature branches, no direct commits on main.
+- Perform tests and code reviews before merges into main, not afterwards.
```

Comment:
Is there an action for running the tests on push? Perhaps we could make one and instruct the users to set up that action on top of firebase

Reply:
cc @eldu